### ISSUE TITLE
Hazard overlay should have the option to skip the overlay with the segmented graph

### ISSUE THREAD

**Issue by LiekeMeijer** [Deltares/ra2ce#321]
In hazard overlay.py under step 3, the code tries to find self.graph_files.base_network_hazard.file (I believe from line 750 onwards). This is an overlay with the segmented graph (100m segments) instead of the more simplified graph. The hazard overlay with the segmented graph takes very long. This functionality should be an option because it is required for the damages analysis. However, it should not be there per default, as it takes very long to run. 

We should have the segmented graph as intermediate output in the output_graph folder. But the hazard overlay should not automatically happen with this segmented graph. 

In short:

1. We want to keep base_graph_hazard_edges and base_graph_hazard_nodes
2. We want to keep base_network.gpkg and base_network. feather
3. We want to have base_network_hazard optional for the damages analysis and not be the default due to high computational time.

The issue can be reproduced with the mozambique data and oost_afrika_prep_mozambique notebook in my postbox.

**Comment by LiekeMeijer** [comment]
When testing the functionalities of hazard overlay, I noticed that the hazard overlay is per default also done for the segmented graph, but this result is not saved. So we now have an overlay that 'slows' the process down, but is then not getting spit out for the user to use.
